pkg/services/authz/rbac/service.go

```diff
@@ -556,25 +566,37 @@ func (s *Service) listPermission(ctx context.Context, scopeMap map[string]bool,
 	defer span.End()
 	ctxLogger := s.logger.FromContext(ctx)
 
-	folderMap, err := s.buildFolderTree(ctx, req.Namespace)
-	if err != nil {
-		ctxLogger.Error("could not build folder and dashboard tree", "error", err)
-		return nil, err
+	t, ok := s.mapper.translation(req.Group, req.Resource)
+	if !ok {
+		ctxLogger.Error("unsupport resource", "group", req.Group, "resource", req.Resource)
+		return nil, status.Error(codes.NotFound, "unsupported resource")
+	}
+
+	var folderMap map[string]FolderNode
+	if t.folderSupport {
+		var err error
+		folderMap, err = s.buildFolderTree(ctx, req.Namespace)
+		if err != nil {
+			ctxLogger.Error("could not build folder and dashboard tree", "error", err)
+			return nil, err
+		}
 	}
 
 	folderSet := make(map[string]struct{}, len(scopeMap))
-	dashSet := make(map[string]struct{}, len(scopeMap))
+
+	prefix := t.prefix()
+	itemSet := make(map[string]struct{}, len(scopeMap))
 	for scope := range scopeMap {
 		if strings.HasPrefix(scope, "folders:uid:") {
-			identifier := scope[len("folders:uid:"):]
+			identifier := strings.TrimPrefix(scope, "folders:uid:")
 			if _, ok := folderSet[identifier]; ok {
 				continue
 			}
 			folderSet[identifier] = struct{}{}
 			getChildren(folderMap, identifier, folderSet)
-		} else if strings.HasPrefix(scope, "dashboards:uid:") {
-			identifier := scope[len("dashboards:uid:"):]
-			dashSet[identifier] = struct{}{}
+		} else {
```

Comment:
Are we 100% sure we don't have a case where a given action applies to multiple kinds?
```suggestion
		} else if strings.HasPrefix(scope, prefix)  {
```

Reply:
For the ones we have in https://github.com/grafana/grafana/pull/99357/files#diff-82d8779ed8336b462284a14970f36d129ff26bc1fd9b8b487b3acb73ed904324R57-R65 yeah we don't have any other scopes